nixos/tests/pam.nix

```diff
@@ -0,0 +1,28 @@
+let
+  name = "pam";
+in
+import ./make-test-python.nix ({ ... }: {
+
+  nodes.machine = { ... }: {
+    imports = [ ../modules/profiles/minimal.nix ];
+
+    krb5.enable = true;
+
+    users = {
+      mutableUsers = false;
+      users = {
+        user = {
+          isNormalUser = true;
+        };
+      };
+    };
+  };
+
+  testScript = ''
+    machine.wait_for_unit("default.target")
+
+    machine.wait_for_file("/etc/pam.d/chfn")
+    with open("/etc/pam.d/chfn") as file_handle:
```

Comment:
AFAIK this doesn't work. It is not like Python has access to the VM that is running the tests, so you need to interact it with the methods from the `machine` object.

For example:

```
machine.succeed("cat /etc/pam.d/chfn | grep 'something'")
```

And yeah, AFAIK you're limited to shell here sadly.

Reply:
D'oh! @jtojnar on Discord (not sure if it's the same person here) had a great idea: `machine.succeed("cat /etc/pam.d/chfn")`, which makes the contents available to the Python script.